skalenetwork/skaled
# Enhancement/sync node broadcast txs

# Enabling transactions broadcast for sync nodes
This PR enables functionality for broadcasting transactions from sync/archive nodes to the core nodes. 

Main changes:   
- Turning on `eth_sendTransaction` and `eth_sendRawTransaction` APIs on sync/archive nodes
- Enabling transaction broadcasting from sync/archive nodes to validator nodes

## Testing
1. Add unit test `send_raw_tx_sync` that is sending tx to sync node and checking that it is appeared in transaction queue
2. Run end-to-end test on devnet:
   - Spin up 4-node chain + sync node 
   - Connect Remix to sync node endpoint
   - Write simple `Simple` contract and deploy it through Remix
   - Send a `store(10)` tx -> tx should pass
   - Check tx receipt on core nodes
   - Make a `retrieve` call -> should return 10


Contract for E2E testing:
```
// SPDX-License-Identifier: GPL-3.0

pragma solidity >=0.7.0 <0.9.0;

/**
 * @title Storage
 * @dev Store & retrieve value in a variable
 */
contract Storage {

    uint256 number;

    /**
     * @dev Store value in variable
     * @param num value to store
     */
    function store(uint256 num) public {
        number = num;
    }

    /**
     * @dev Return value 
     * @return value of 'number'
     */
    function retrieve() public view returns (uint256){
        return number;
    }
}
```
